k2/csrc/fsa_utils.cu

```diff
@@ -177,7 +227,22 @@ static Fsa TransducerFromStream(std::string first_line, std::istringstream &is,
     }
   } while (std::getline(is, line));
 
-  K2_CHECK(is) << "Failed to read fsa from string";
+  // Post processing on final states. When openfst is true, we may have multiple
+  // final states with weights associated with them. We will have to add a super
+  // final state, and convert that into the K2 format (final state with no
+  // weight).
+  if (original_final_states.size() > 0) {
+    K2_CHECK_EQ(openfst, true);
+    K2_CHECK_EQ(original_final_states.size(), original_final_weights.size());
+    int32_t super_final_state = max_state + 1;
+    for (int32_t i = 0; i != original_final_states.size(); ++i) {
```

Comment:
I would recommend to use either `std::size_t i = 0` or  `static_cast` of `size()`
to avoid compiler warnings about comparison between signed and unsigned integers.

----

An alternative is

```cpp

for (auto i : original_final_states) {
   // ....
}

Reply:
Fixed.